### PR TITLE
chore: add test for compute_calls_instructions_len

### LINKED ISSUE #1129: test: test `compute_call_instructions_len`
This function is currently untested, even though the slightest modification of it can lead to the `ContractNotFound` issue.

### PR DESCRIPTION
**Pull request by Br1ght0ne**
Close #1129.

I added a test case for an empty (random) contract, would like some feedback on what kinds of non-empty contracts to test.

### Checklist
- [x] I have linked to any relevant issues.
- [x] I have updated the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by Br1ght0ne** [comment]
@iqdecay Thank you for the pointers - will do.

**Review by MujkicA — DISMISSED** [review]
LGTM

File: packages/fuels-programs/src/call_utils.rs [hunk 1]
```diff
@@ -920,4 +920,84 @@ mod test {
...
+        }
+
+        #[test]
+        fn test_with_gas_offset() {
+            let mut call = ContractCall::new_with_random_id();
+            call.call_parameters = call.call_parameters.with_gas_forwarded(0);
+            let instructions_len = compute_calls_instructions_len(&[call]).unwrap();
+            assert_eq!(
+                instructions_len,
+                Instruction::SIZE * (BASE_INSTRUCTION_COUNT + 2)
```
Comment (digorithm) [inline packages/fuels-programs/src/call_utils.rs:946]: We should use this opportunity to have these tests act as documentation for the inner workings of the SDK/client. Why `+ 2` here and `+ 4` down there? We should have this info as comments right here.

**Review by digorithm — CHANGES_REQUESTED** [review]
LGTM -- just a minor documentation-related nit.

File: packages/fuels-programs/src/call_utils.rs [hunk 2]
```diff
@@ -920,4 +920,84 @@ mod test {
...
+            );
+        }
+
+        #[test]
+        fn test_with_enum_with_heap_type_variants() {
+            let mut call = ContractCall::new_with_random_id();
+            call.output_param = ParamType::Enum {
+                variants: EnumVariants::new(vec![
+                    ParamType::Vector(Box::new(ParamType::U8)),
+                    ParamType::Vector(Box::new(ParamType::B256)),
```
[the remark this answers is not in the document]
Comment (Br1ght0ne) [reply packages/fuels-programs/src/call_utils.rs:991]: Thank you so much for pointing out the test cases!

**Review by hal3e — COMMENTED** [review]
This looks very good. Left some small suggestions.

**Review by MujkicA — APPROVED** [review]
Nice

**Review by hal3e — APPROVED** [review]
Thanks!

**Pull request merged**